### PR TITLE
Updates to RTD documentation

### LINKED ISSUE #2395: Regular RTD documentation update
### What new functionality do you need?

Updated RTD documentation that now includes recent additions to staged IC sets and new platforms (AWS). Also make some additional general updates.

### What are the requirements for the new functionality?

Up-to-date documentation.

### Acceptance Criteria

Updated documentation includes recent changes.

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
# Description

This PR makes some regular updates to the RTD documentation. This includes:

- Textual updates
- Better definition of GDA subfolder structure
- Added note about GDAur having been discontinued
- Adjust copyright and author information
- Fix Git version table and update its contents
- Add "Table of Contents" header before table of contents on front page
- Add AWS ICs path to init page
- Add link to UFS_UTILS gdas_init RTD documentation
- Add note about bash to `gw_setup.sh` section and added a warning block

Resolves #2395 

# Type of change

- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? YES

# How has this been tested?

Reviewed documentation artifacts after changes were made.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -13,13 +13,14 @@
...
-
+from datetime import datetime
 
 # -- Project information -----------------------------------------------------
 
 project = 'Global-workflow'
-copyright = '2023, Kate Friedman, Walter Kolczynski, Rahul Mahajan, Lin Gan, Arun Chawla'
-author = 'Kate Friedman, Walter Kolczynski, Rahul Mahajan, Lin Gan, Arun Chawla'
+year = datetime.now().year
+copyright = f"2015-{year} NOAA/NWS/NCEP/EMC, Kate Friedman, Walter Kolczynski, Rahul Mahajan, Lin Gan"
```
Comment (aerorahul) [inline docs/source/conf.py]:
```suggestion
copyright = f"2015-{year} NOAA/NWS/NCEP/EMC"
```
I don't think the individuals can hold any copyright (besides, they wouldn't want to for legal reasons)

File: docs/source/conf.py [hunk 2]
```diff
@@ -13,13 +13,14 @@
...
+author = 'Kate Friedman, Walter Kolczynski, Rahul Mahajan, Lin Gan'
```
Comment (aerorahul) [inline docs/source/conf.py]:
```suggestion
author = 'Kate Friedman, Walter Kolczynski, Rahul Mahajan, Lin Gan, and numerous collaborators and contributors'
```

**Review by aerorahul — APPROVED** [review]
looks good.

**Review by HenryRWinterbottom — APPROVED** [review]
LGTM.

**Pull request merged**